### ISSUE TITLE
bug in equality of complexes (really equality of zero modules)

### ISSUE THREAD

**Issue by mahrud** [Macaulay2/M2#3346]
Here's an unfortunate situation that happened today:
```m2
restart
needsPackage "Complexes"

S = kk[a..d]
psi = map(kk, S)
I = ideal"ab, ad, bc, c3"
F = freeResolution comodule I

C = psi complex applyValues(F.dd.map, basis_0)
D = part(0, F)
prune C == D -- true
C == D       -- false!!!!!!
```
Looking at the output, you might guess that it's because concentrations aren't equal:
```m2
i6 : C = psi complex applyValues(F.dd.map, basis_0)

o6 = image | 1 | <-- image 0 <-- image 0 <-- image 0
                                              
     0               1           2           3

o6 : Complex

i7 : D = part(0, F)

       1
o7 = kk
      
     0

o7 : Complex
```
But in reality it's because M2 treats the zero submodule in different ambient modules to be different:
```m2
M = source C.dd_1
N = source D.dd_1
M == 0 -- true
N == 0 -- true
M == N -- false
```
And this is why the modules are considered not the same:
```m2
i12 : M = source C.dd_1

o12 = image 0

                                4
o12 : kk-module, submodule of kk

i13 : N = source D.dd_1

o13 = 0

o13 : kk-module
```
I can imagine the argument that these two modules should not be equal, but I think the equality of the complexes above failing is obviously a bug. So I would like to know the arguments for having these modules be unequal. Does it break something? Should we just change them to be equal?

I personally think they should be equal, keeping in mind that they are not === equal.

cc: @mikestillman

**Comment by mahrud** [comment]
Actually, here's another way to see this:
```m2
i1 : M = image map(QQ^4, QQ^0, 0)

o1 = image 0

                               4
o1 : QQ-module, submodule of QQ

i2 : prune M

o2 = 0

o2 : QQ-module

i3 : ooo == oo

o3 = false
```

**Comment by mahrud** [comment]
This just leads to too many unexpected results:
```m2
i1 : needsPackage "Complexes";

i2 : S = QQ[x,y];

i3 : C = complex(0 * S^{-1})

o3 = image 0
      
     0

o3 : Complex

i4 : D = part(0, C)

o4 = 0
      
     0

o4 : Complex

i5 : C == D

o5 = false

i6 : part(-1, C) == D

o6 = true
```

**Comment by mahrud** [comment]
Fixed in 75cb5cc9fe5743518994d4f5d6fb2705b3b66854. Hasn't broken anything yet! 🤞
